github-actions/feature-request/src/octokit.ts

```diff
@@ -0,0 +1,259 @@
+    return this._octokit;
+  }
+
+  protected mockLabels: Set<string> = new Set();
+
+  constructor(
+    protected token: string,
+    protected params: { repo: string; owner: string },
+    protected options: { readonly: boolean } = { readonly: false },
+  ) {
+    this._octokit = new GitHub(token);
+  }
+
+  async *query(query: Query): AsyncIterableIterator<GitHubIssueAPI> {
+    const q = query.q + ` repo:${this.params.owner}/${this.params.repo}`;
+
+    const options = this.octokit.search.issuesAndPullRequests.endpoint.merge({
+      ...query,
+      q,
+      per_page: 100,
+      // To access reactions we need a specific media type we specify via the accept header
+      // https://docs.github.com/en/rest/reference/repos#list-commit-comments-for-a-repository-preview-notices
+      headers: { Accept: 'application/vnd.github.squirrel-girl-preview+json' },
+    });
+
+    let pageNum = 0;
+
+    const timeout = async () => {
+      if (pageNum < 2) {
+        /* pass */
+      } else if (pageNum < 4) {
+        await new Promise((resolve) => setTimeout(resolve, 10000));
+      } else {
+        await new Promise((resolve) => setTimeout(resolve, 30000));
+      }
+    };
+
+    for await (const pageResponse of this.octokit.paginate.iterator(options)) {
+      await timeout();
+      const page: Array<Octokit.SearchIssuesAndPullRequestsResponseItemsItem> = pageResponse.data;
+      log(`Page ${++pageNum}: ${page.map(({ number }) => number).join(' ')}`);
+      for (const issue of page) {
+        yield new OctoKitIssue(
+          this.token,
+          this.params,
+          this.octokitIssueToIssue(issue),
+          this.options,
+        );
+      }
+    }
+  }
+
+  async isOrgMember(name: string, org: string): Promise<boolean> {
+    if (this._orgMembers.size) {
+      return this._orgMembers.has(name);
+    }
+
+    const response = this.octokit.paginate.iterator(
+      this.octokit.orgs.listMembers.endpoint.merge({
+        org,
+        per_page: 100,
+      }),
+    );
+
+    for await (const page of response) {
+      for (const user of page.data as Octokit.OrgsListMembersResponse) {
+        this._orgMembers.add(user.login);
+      }
+    }
+
+    return this._orgMembers.has(name);
+  }
+
+  protected octokitIssueToIssue(
+    issue: Octokit.IssuesGetResponse | Octokit.SearchIssuesAndPullRequestsResponseItemsItem,
+  ): Issue {
+    return {
+      author: { name: issue.user.login, isGitHubApp: issue.user.type === 'Bot' },
+      body: issue.body,
+      number: issue.number,
+      title: issue.title,
+      labels: (issue.labels as Octokit.IssuesGetLabelResponse[]).map((label) => label.name),
+      open: issue.state === 'open',
+      locked: (issue as any).locked,
+      numComments: issue.comments,
+      reactions: (issue as any).reactions,
+      assignee: issue.assignee?.login ?? (issue as any).assignees?.[0]?.login,
+      createdAt: +new Date(issue.created_at),
+      updatedAt: +new Date(issue.updated_at),
```

Comment:
Is it possible that the issue was never updated (should we have a check similar to `closed_at` one below)?

Reply:
That's a good point! Based on octokit's typings, `updated_at` is always a string, so we shouldn't ever get a `null` value.

Reply:
The value is always presented because on creation it becomes updated_at the same time as creation.